app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -47,6 +47,12 @@ def deployment_spec(opts)
       target[SPEC_KEY_MAPPING[r]] = resource.ems_ref if resource.present?
     end
 
+    if target["folder_id"].blank? && opts[:datacenter_id].present?
+      dc = Datacenter.find_by(:id => opts[:datacenter_id])
```

Comment:
If https://github.com/ManageIQ/manageiq-providers-vmware/pull/680/files#r543462499 is the case (aka datacenter_id is the _vmware api_ id) then we can't `find_by(:id)` here we would have to `find_by(:ems_ref)`

Reply:
And if we're searching by ems_ref we need to have the query scoped to the current EMS so we don't find another vcenter's "datacenter-1"

Reply:
`opts[:datacenter_id]` is our `datacenter_id` in VMDB which will be selected when adding a new OVF service template.
UI work will come after these PRs. cc @h-kataria

When adding a new OVF service template, if the user selects a datacenter but not selects any folder, then the VM folder for the selected datacenter would be used for the OVF deployment.

Reply:
Yup I missed that you were checking both `target` and `opts`